File: app/src/main/java/empire/stark/firststep/common/activity/BaseActivityModule.java
package empire.stark.firststep.common.activity;

import dagger.Module;
import dagger.Provides;
import empire.stark.firststep.data.YenMinh;
import empire.stark.firststep.main.SubComponentMainActivity;

/**
 * Created by YEN_MINH on 3/23/2017 3:45 AM.
 * Code and Life ( ^ .,,. ^ )
 * ---------------------------------------------
 */

@Module
public class BaseActivityModule {
    @Provides
    public YenMinh yenMinh() {
        return new YenMinh("minh123", "Minh");
    }

}

File: app/src/main/java/empire/stark/firststep/common/dagger/BaseActivityModule.kt
package empire.stark.firststep.common.dagger

import android.app.Activity

import dagger.Binds
import dagger.Module
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap
import empire.stark.firststep.cat.CatActivity
import empire.stark.firststep.cat.dagger.CatActivitySubComponent
import empire.stark.firststep.common.BaseActivity
import empire.stark.firststep.main.view.MainActivity
import empire.stark.firststep.main.dagger.MainActivitySubComponent


/**
 * Created by YEN_MINH on 3/9/2017.
 */
@Module(subcomponents = arrayOf(SubComponentBaseActivity::class, MainActivitySubComponent::class, CatActivitySubComponent::class))
abstract class BaseActivityModule {

    @Binds
    @IntoMap
    @ActivityKey(BaseActivity::class)
    abstract fun bindActivityInjectorFactory(builder: SubComponentBaseActivity.Builder): AndroidInjector.Factory<out Activity>

    @Binds
    @IntoMap
    @ActivityKey(MainActivity::class)
    abstract fun bindMainActivityInjectorFactory(builder: MainActivitySubComponent.Builder): AndroidInjector.Factory<out Activity>

    @Binds
    @IntoMap
    @ActivityKey(CatActivity::class)
    abstract fun bindCatActivityInjectorFactory(builder: CatActivitySubComponent.Builder): AndroidInjector.Factory<out Activity>

}
